### PR TITLE
Add custom pans

### PR DESCRIPTION
**Pull request by smkuhne**
This PR aims at closing #150 by supporting the creation of a CustomCardIssuer as well as the creation a PanFormatters aimed specifically at new types of cards. Support for custom cards is being added on top of the default supporting cards.

### CODE REVIEW

File: scan-payment/src/main/java/com/getbouncer/scan/payment/card/CardIssuer.kt [hunk 1]
```diff
@@ -14,6 +14,11 @@ sealed class CardIssuer(val displayName: String) {
     object Visa : CardIssuer("Visa")
 }
 
+/**
+ * A data class for custom card issuers
+ */
+data class CustomCardIssuer(override val displayName: String) : CardIssuer(displayName)
```
Comment (awushensky) [inline scan-payment/src/main/java/com/getbouncer/scan/payment/card/CardIssuer.kt:20]: nit: relocate this into `CardIssuer` and rename it to just `Custom`

File: scan-payment/src/main/java/com/getbouncer/scan/payment/card/PaymentCardUtils.kt [hunk 2]
```diff
@@ -217,14 +223,31 @@ fun String.lastFour(): String = lastFourFromPan(this)
  */
 @CheckResult
 internal fun getIssuerData(cardNumber: String): IssuerData? =
-    ISSUER_TABLE.firstOrNull { iinFromPan(cardNumber).toInt() in it.iinRange }
+    CUSTOM_ISSUER_TABLE.firstOrNull { iinFromPan(cardNumber).toInt() in it.iinRange }
+        ?: ISSUER_TABLE.firstOrNull { iinFromPan(cardNumber).toInt() in it.iinRange }
```
Comment (awushensky) [inline scan-payment/src/main/java/com/getbouncer/scan/payment/card/PaymentCardUtils.kt:227]:
```suggestion
    (CUSTOM_ISSUER_TABLE + ISSUER_TABLE).firstOrNull { iinFromPan(cardNumber).toInt() in it.iinRange }
```

File: scan-payment/src/main/java/com/getbouncer/scan/payment/card/PaymentCardUtils.kt [hunk 3]
```diff
@@ -217,14 +223,31 @@ fun String.lastFour(): String = lastFourFromPan(this)
...
  */
 @CheckResult
-private fun getIssuerData(issuer: CardIssuer): List<IssuerData> =
-    ISSUER_TABLE.filter { it.issuer == issuer }
+private fun getIssuerData(issuer: CardIssuer): List<IssuerData> {
+    val mutualIssuerData = CUSTOM_ISSUER_TABLE.filter { it.issuer == issuer }.toMutableList()
+    mutualIssuerData.addAll(ISSUER_TABLE.filter { it.issuer == issuer })
+
+    return mutualIssuerData.toList()
+}
```
Comment (awushensky) [inline scan-payment/src/main/java/com/getbouncer/scan/payment/card/PaymentCardUtils.kt:238]:
```suggestion
private fun getIssuerData(issuer: CardIssuer): List<IssuerData> =
    (CUSTOM_ISSUER_TABLE + ISSUER_TABLE).filter { it.issuer == issuer }
```

File: scan-payment/src/main/java/com/getbouncer/scan/payment/card/PaymentCardUtils.kt [hunk 4]
```diff
@@ -217,14 +223,31 @@ fun String.lastFour(): String = lastFourFromPan(this)
...
+/**
+ * Adds support for a new [CardIssuer]
+ */
+@CheckResult
+fun supportCardIssuer(
+    validIins: IntRange,
+    validCardIssuer: CardIssuer,
+    validPanLengths: List<Int>,
+    validCvcLengths: List<Int>,
+    validationFunction: PanValidator = LengthPanValidator + LuhnPanValidator
```
Comment (awushensky) [inline scan-payment/src/main/java/com/getbouncer/scan/payment/card/PaymentCardUtils.kt:249]: nit: drop the `valid`

File: scan-payment/src/main/java/com/getbouncer/scan/payment/card/PaymentCardUtils.kt (hunk 3 above)
Comment (smkuhne) [reply scan-payment/src/main/java/com/getbouncer/scan/payment/card/PaymentCardUtils.kt:238]: Was looking for solution like this, but didn't find any matching functions, the answer was way easier than I thought.

**Pull request merged**